### PR TITLE
add dnssec constraint template

### PR DESCRIPTION
**Pull request by tangyan**

### CODE REVIEW

File: validator/dnssec.rego [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	asset := input.asset
+	asset.asset_type == "dns.googleapis.com/ManagedZone"
+
+	asset.resource.data.dnssecConfig.state == "OFF"
```
Comment (t12g) [inline validator/dnssec.rego]: can we compare it to not on instead? there's a 3rd "transfer" state

File: validator/test/fixtures/constraints/require_dnssec/data.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: constraints.gatekeeper.sh/v1alpha1
+kind: GCPDNSSECEnabledConstraintV1
+metadata:
+  name: require_dnssec
+spec:
+  constraintVersion: 0.1.0
```
Comment (t12g) [inline validator/test/fixtures/constraints/require_dnssec/data.yaml]: let's delete this line; it's no longer in spec

File: validator/dnssec_test.rego [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+	violation := issues[_]
+}
+
+# Confirm total violations count
+test_dnssec_violations_count {
+	count(all_violations) == 1
+}
+
+test_dnssec_violations_basic {
```
Comment (t12g) [inline validator/dnssec_test.rego]: can we add the sample data for the transfer case and check it?

Comment (tangyan) [reply validator/dnssec_test.rego]: Done

File: validator/dnssec.rego (hunk 1 above)
Comment (tangyan) [reply validator/dnssec.rego]: Done

File: validator/test/fixtures/constraints/require_dnssec/data.yaml (hunk 2 above)
Comment (tangyan) [reply validator/test/fixtures/constraints/require_dnssec/data.yaml]: Done

**Pull request merged**